### DISCUSSION TITLE
Problem with new date picker

### DISCUSSION THREAD

**Discussion by pakricard** [timschofield/webERP#268] [Development Discussion & Specification]
Hi: 
I am updating my install to use the new date picker as detailed in https://github.com/timschofield/webERP/wiki/DatePicker, to move faster to v5

Using the newest files from GitHub (commit https://github.com/timschofield/webERP/commit/c00d84ea82e7501ae38f0bcc3d2018220dffcb12):

- AuditTrail.php
- AddCustomerNotes.php
- css folder
- javascripts folder
- includes/MiscFunctions.php
- includes/DateFunctions.php

$_SESSION['DefaultDateFormat'] = 'd/m/Y'
$_SESSION['MonthsAuditTrail'] = 2

If today (02 December 2024) I run the newest AuditTrail.php

![image](https://github.com/user-attachments/assets/c542411b-d9ed-4dfd-b32c-37777c6f11a7)

It shows the dates in format m/d/Y which is incorrect

To verify the settings, I modified the if structure in line 16 of AuditTrail.php to read as:

```
if (!isset($_POST['FromDate'])){
	$_POST['FromDate'] = Date($_SESSION['DefaultDateFormat'],mktime(0,0,0, Date('m')-$_SESSION['MonthsAuditTrail']));
	prnMsg($_SESSION['DefaultDateFormat']);
	prnMsg($_POST['FromDate']);
}

```
And the prnMsg() show the correct values:

![image](https://github.com/user-attachments/assets/98305d0e-e20f-4de9-ac09-31101cc49d2a)

But, somehow, the echo on line 48 shows the wrong format.

Moreover: If I want to use the date picker and change the from date to, let's say 15th October I get the following message:

![image](https://github.com/user-attachments/assets/6d4f2883-9717-4d12-9118-640719bbb8b5)
 
This behaviour is happening to all the scripts using the new date pciker, as example AddCustomerNotes.php

![image](https://github.com/user-attachments/assets/245b574d-e0de-44cf-bf11-327a463647b9)

It shows the date picker in the wrong format m/d/Y, once the note is inserted it shows it on the screen as m/d/Y, but it is stored correctly on the database.

![image](https://github.com/user-attachments/assets/05ae0bcb-d701-442c-a86f-2e4719e1545b)

As I mentioned on issue #242, it does work OK on the demo webERP site. I checked every possibility I could think of, but I am completely lost. 

Is it only me having these weird issues?
Any other file or setting I should update from GitHub that influences the behaviour of the date picker?

Thanks, 
Ricard

**Comment by timschofield** [comment]
This is a strange one. According to the specification the HTML5 date
picker will display the browser's locale date format (maybe not
optimal, but let's set that aside for the moment) but always return a
date in the format 'YYYY-mm-dd' to the server. On my machine here I
have 3 browsers - Firefox, Chromium, and Opera. Firefox and Chromium
work exactly as expected (displaying as 'dd/mm/YYYY'), but Opera
displays mm/dd/YYYY. I have double checked every setting in Opera, and
it appears to have the correct locale, but the display is wrong.

The 2 commonly used browsers I don't have are Edge, and Safari. Does
anybody use these, and if so do they work correctly?

This is my fault as I should have researched this more thoroughly.
Problem is, it was a big change and I'm reluctant to change this, if
we can work around it.

Ricard, what browser are you using?

Tim


On Mon, 2 Dec 2024 at 07:59, pakricard ***@***.***> wrote:
>
> Hi:
> I am updating my install to use the new date picker as detailed in https://github.com/timschofield/webERP/wiki/DatePicker, to move faster to v5
>
> Using the newest files from GitHub (commit c00d84e):
>
> AuditTrail.php
> AddCustomerNotes.php
> css folder
> javascripts folder
> includes/MiscFunctions.php
> includes/DateFunctions.php
>
> $_SESSION['DefaultDateFormat'] = 'd/m/Y'
> $_SESSION['MonthsAuditTrail'] = 2
>
> If today (02 December 2024) I run the newest AuditTrail.php
>
> image.png (view on web)
>
> It shows the dates in format m/d/Y which is incorrect
>
> To verify the settings, I modified the if structure in line 16 of AuditTrail.php to read as:
>
> if (!isset($_POST['FromDate'])){
> $_POST['FromDate'] = Date($_SESSION['DefaultDateFormat'],mktime(0,0,0, Date('m')-$_SESSION['MonthsAuditTrail']));
> prnMsg($_SESSION['DefaultDateFormat']);
> prnMsg($_POST['FromDate']);
> }
>
> And the prnMsg() show the correct values:
>
> image.png (view on web)
>
> But, somehow, the echo on line 48 shows the wrong format.
>
> Moreover: If I want to use the date picker and change the from date to, let's say 15th October I get the following message:
>
> image.png (view on web)
>
> This behaviour is happening to all the scripts using the new date pciker, as example AddCustomerNotes.php
>
> image.png (view on web)
>
> It shows the date picker in the wrong format m/d/Y, once the note is inserted it shows it on the screen as m/d/Y, but it is stored correctly on the database.
>
> image.png (view on web)
>
> As I mentioned on issue #242, it does work OK on the demo webERP site. I checked every possibility I could think of, but I am completely lost.
>
> Is it only me having these weird issues?
> Any other file or setting I should update from GitHub that influences the behaviour of the date picker?
>
> Thanks,
> Ricard
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>



-- 
www.weberp.org
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Comment by dalers** [comment]
Recent version of Edge on Windows 10 Pro ("Version 131.0.2903.70 (Official build) (64-bit)") (our managed IT services provider decides when we update).

First, where is AuditTrail in the menu? I would have expected it to be available to an Administrator under Utilities.

1. webERP.org demo - I logged into weberp.org/demo and accessed AuditTrail.php. The from-date legend in the browser is "From Date d/m/Y"

![image](https://github.com/user-attachments/assets/1bb28698-e0e4-483a-b723-47a7da602911)

2. _my_ demo website with yesterday's[1] code - I logged in and accessed AuditTrail.php . The legend in the browser is "From Date Y-m-d" (different from weberp.org/demo), but in both cases the picker is showing "mm/dd/yyyy".

![image](https://github.com/user-attachments/assets/0a84ebe8-4026-42ba-b5ae-e1585a4b46c6)


Something odd is happening...... maybe browsers have defaults and a desired format needs to be made explicit somehow....


[1] commit c00d84ea82e7501ae38f0bcc3d2018220dffcb12 (HEAD -> master, origin/master, origin/HEAD)
Merge: 63a42f372 81449ae93
Author: Dale Scott <dale@dalescott.net>
Date:   Sun Dec 1 10:39:35 2024 -0700

**Comment by dalers** [comment]
weberp.org/demo vs latest code using Chrome, which matches Edge. The date pickers show today's date in MM/DD/YYYY but the format of the "from" and "to" date _legends_ changes depending on browser.

![image](https://github.com/user-attachments/assets/1f87dd79-9eee-482a-ac98-d0db3735446b)

**Comment by dalers** [comment]
However an Audit Trail report appears to be generated correctly in both Edge and Chrome after using the date picker to change dates.

![image](https://github.com/user-attachments/assets/ea9b8f23-c5c6-48a5-9c3d-011dfddb8146)

**Comment by pakricard** [comment]
I use Chrome, Firefox or Edge on Windows 11 Pro. Everything using latest updates, patches, etc.

- On Chrome, I have the weird behaviour of m/d/Y.
- Firefox and Edge are showing the right d/m/Y

On my Android tablet using Chrome it works OK

weberp.org demo and Chrome shows it wrong, Firefox and Edge OK

So, it seems now it is browser and user-settings dependant, as Edge is working OK for me but not for @dalers 

I played with Chrome settings and the only way to make it work, was to delete English as a language (the default one AND the US version) and use a language that has d/m/Y as the regular date format and it worked.
English Australia, Catalan, Spanish or Indonesian (as primary Chrome languages) made the trick.

Could you please check if changing the primary language of your browser to anything except US English does the trick?
